packages/api/src/external/carequality/process-outbound-patient-discovery-resps.ts

```diff
@@ -62,11 +70,57 @@ export async function processOutboundPatientDiscoveryResps({
       },
     });
 
-    log(`Completed.`);
+    const rerunPdOnNewDemographics = cqData?.rerunPdOnNewDemographics;
+    const facilityId = cqData?.discoveryFacilityId;
+    let foundNewDemographics = false;
+    if (rerunPdOnNewDemographics) {
+      foundNewDemographics = checkForNewDemographics(patient, cqLinks);
+    }
+
+    let scheduledPdRequest = cqData?.scheduledPdRequest;
+    if (foundNewDemographics && facilityId) {
+      const updatedPatient = await scheduleOrRunPatientDiscovery({
```

Comment:
Why is `discoveryFacilityId` optional?

Reply:
It's optional in the patient `PatientDataCommonwell` and `PatientDataCarequality` because it may not be set. I could wrap the patient discovery values (i.e. requestId, facilityId, startedAt, augmentedPatient, etc.) in a object, in which case it wouldn't be optional within that object.

Reply:
TODO: Need to consider passing the state values to the results function explicitly like requestId. Right now it's a bit of a pain to use the patient object for this kind of stuff in CQ. I also don't want to introduce any new models just for this... although maybe `cq_patient_data` and `cw_patient_data` can be leveraged.

Reply:
Add the TODO into the code -- I'd like to create an issue if other people agree. Generally, if we can pass data explicitly into the lambda (like we do with requestId), we should pass in *all* used info (so include startedAt, facilityId, and rerunOnNewDemographics).

I didn't want to start editing lambda inputs, etc. for this PR, so we need to perform these checks.